packages/components/src/components/ViewerWithMenuBar/index.tsx

```diff
@@ -28,14 +32,20 @@ export const ViewerWithMenuBar = forwardRef<SquiggleViewerHandle, Props>(
       editor,
       defaultTab,
     },
-    viewerRef
+    viewerWithMenuBarRef
   ) {
     const [viewerTab, setViewerTab] = useState<ViewerTab>(
       defaultTab ?? defaultViewerTab(simulation.output)
     );
 
+    const viewerRef = useRef<SquiggleViewerHandle>(null);
     const { output } = simulation;
 
+    useImperativeHandle(viewerWithMenuBarRef, () => ({
+      squiggleViewerHandle: viewerRef.current,
+      setViewerTab,
```

Comment:
I can see why exposing `setViewerTab` can be useful in general, but for "focus on path" actions it makes more sense to move the logic to this component.

Like, define the handle for this component as:
```typescript
{
  ...viewerRef.current,
  focusFromPath: (path) => {
    ...; // change mode
    viewerRef.current?.focusFromPath(path);
  }
}
```

Reply:
I thought about that. Felt like it was kind of weird that the ``ViewerWithHeader`` component was doing the work of doing some of this logic. Happy to change though, feel neutral about it.

Reply:
> Felt like it was kind of weird that the `ViewerWithHeader` component was doing the work of doing some of this logic.

Well, the specific way it splits the output into tabs is its area of responsibility, and it could grow quite complex over time (e.g. imagine if it had "pins" or even "temporary named sets of pins" that appear in the dropdown as you add them; you won't be able to tell from the outside which tab to use).